src/test/netbase_tests.cpp

```diff
@@ -156,6 +156,42 @@ BOOST_AUTO_TEST_CASE(embedded_test)
     BOOST_CHECK_EQUAL(addr1.ToStringAddr(), addr2.ToStringAddr());
 }
 
+BOOST_AUTO_TEST_CASE(dns_lookup)
+{
+    // Non-numeric lookup that does not require actual DNS service
+    BOOST_CHECK(LookupHost("localhost", /*fAllowLookup=*/true).has_value());
+    BOOST_CHECK(!LookupHost("fail.doesnotexist", /*fAllowLookup=*/true).has_value());
```

Comment:
Won't this still make an externally-visible DNS request?

Reply:
Yep you are absoutley right, I'll remove.